### PR TITLE
Increase duration between polling from stdin

### PR DESCRIPTION
**Pull request by eatkins**
I was testing the cpu utilization of sbt when idle and noticed that it
was a bit higher than I expected. It was using O(15%) of a cpu on
average. After increasing these periods, the value dropped to O(2%).

- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: main/src/main/scala/sbt/internal/Idle.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+package sbt.internal
+
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+private[sbt] object Idle {
+  val value: FiniteDuration =
+    try System.getProperty("sbt.idle.seconds", "300").toInt.seconds
+    catch { case NonFatal(_) => 5.minute }
+  val idleTimeout: FiniteDuration = 500.millis
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/internal/Idle.scala:17]: So in the worst case, it will add 0.5 second latency after you type `test` and hit the enter key?

**Comment by smarter** [comment]
Why does sbt need polling to read stdin ?

File: main/src/main/scala/sbt/internal/Idle.scala (hunk 1 above)
Comment (eatkins) [reply main/src/main/scala/sbt/internal/Idle.scala:17]: Yeah.

**Comment by eatkins** [comment]
@smarter it is necessary to poll because `InputStream.read` blocks uninterruptibly on java so if you spin up a background thread to read from stdin and write to a queue, you can't shutdown that thread. There is probably a better way to get bytes from stdin, but that's outside of the scope of 1.3.0.

**Comment by eed3si9n** [comment]
> Why does sbt need polling to read stdin ?

`shell` command stops listening to the stdin once it starts processing user entered command. So I tried to implement similar mechanism for commands that came in from sbt server. We can revisit it later.

File: main/src/main/scala/sbt/internal/Idle.scala (hunk 1 above)
Comment (eed3si9n) [reply main/src/main/scala/sbt/internal/Idle.scala:17]: Could we reduce the latency to somewhere low enough that human would not perceive it but high enough that CPU won't mind?

**Comment by eatkins** [comment]
I am working on a better approach to fixing this.

**Pull request closed without merging**